.github/workflows/python-app.yml

```diff
@@ -3,32 +3,39 @@
 
 name: Python application
 
-on: [push, pull_request]
+on:
+  # Trigger the workflow on push or pull request,
+  # but only for the master branch
+  push:
+    branches:
+      - main
```

Comment:
I am not sure we want to just run this on pushes to main, I feel there is value in having a CI ran for each push to development branches. And the jobs are fairly light-weight, so we are not wasting a lot of computation.